mobile/mysterium/openvpn_connection_setup.go

```diff
@@ -85,8 +87,42 @@ func (adapter channelToCallbacksAdapter) OnStats(openvpnStats openvpn3.Statistic
 // Openvpn3TunnelSetup is alias for openvpn3 tunnel setup interface exposed to Android/iOS interop
 type Openvpn3TunnelSetup openvpn3.TunnelSetup
 
-// OverrideOpenvpnConnection replaces default openvpn connection factory with mobile related one
-func (mobNode *MobileNode) OverrideOpenvpnConnection(tunnelSetup Openvpn3TunnelSetup) {
+// ReconnectableSession interface exposing reconnect for an active session
+type ReconnectableSession interface {
+	Reconnect(afterSeconds int) error
+}
+
+type sessionTracker struct {
+	session *openvpn3.Session
```

Comment:
This reference is being set/read from possibly different threads. Some kind of sync mechanism is needed, like Mutex